### PR TITLE
lxd/instances: Add support to retry instance autostart when autostart…

### LINKED ISSUE #8858: Retry failed instance startup when auto-started
We've seen a few times where LXD fails to autostart some instances because of network/storage type issues.
It'd be good to retry startup up to 3 times waiting 5s or so in between tries.

If things still fail, then a persistent warning should be added using our new warnings API so the user can easily see what happened.

### PR DESCRIPTION
**Pull request by zakufish**
Add support to retry instance autostart when autostart fails.

Resolves #8858.

Signed-off-by: Ricky Yu <rickyyu519@protonmail.com>

### CODE REVIEW

File: lxd/instances.go [hunk 1]
```diff
@@ -229,7 +230,32 @@ func instancesRestart(s *state.State) error {
...
-		if shared.IsTrue(autoStart) || (autoStart == "" && lastState == "RUNNING") {
+		if shared.IsTrue(autoStart) {
+			wg.Add(1)
+
+			go func(c instance.Instance) {
+				err := *new(error)
+				for retry := 0; retry < 3; retry ++ {
+					err = c.Start(false)
+					if err != nil {
+						logger.Errorf("Failed to start instance '%s': %v", c.Name(), err)
```
Comment (tomponline) [inline lxd/instances.go:241]: Can you uaw `%q` here instead of `'%s'`

Comment (zakufish) [reply lxd/instances.go]: Done :)

**Comment by stgraber** [comment]
@zakufish Thanks for the branch! Could you edit your commit messages to include the required `Signed-off-by: Firstname Lastname <email>` line? (see contributing.md)

**Comment by zakufish** [comment]
@stgraber Got it. Done :)

**Comment by zakufish** [comment]
Hi @stgraber It's my first time contributing to the open-source project, so I've made quite a mess with the

**Pull request closed without merging**